pkg/util/http/transport.go

```diff
@@ -88,6 +89,14 @@ func CreateHTTPTransport(cfg pkgconfigmodel.Reader) *http.Transport {
 	// desirable side-effect of disabling http/2; if removing those fields then
 	// consider the implication of the protocol switch for intakes and other http
 	// servers. See ForceAttemptHTTP2 in https://pkg.go.dev/net/http#Transport.
+
+	var tlsHandshakeTimeout time.Duration
+	if cfg.IsSet("tls_handshake_timeout") {
+		tlsHandshakeTimeout = cfg.GetDuration("tls_handshake_timeout")
+	} else {
+		tlsHandshakeTimeout = 10 * time.Second
```

Comment:
I agree that for debugging purposes would be useful. Still, adding a new noob for customers play could be problematic. It would require training the support team to know about this new configuration, etc.
The default time 10 seconds is more than enough and at the same time is too much. 

All of that to say that I'm on the defense to add configuration.

What are your thoughts @pgimalac?

Reply:
The config is not documented and we would not advise customers to use it, its only purpose is debugging.
I agree that 10 seconds is already a lot, but being able to change it would help us figure out if it would work with a little bigger duration, or if it's actually always failing even with a huge limit.

We have hundreds of similar undocumented configs that can tweak subtle behaviors of the agent, and they (in theory) only ever used when we tell customers to, so I don't think one more would be an issue.

Reply:
Sounds good my friend. Let's add it 😄